hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -3882,6 +3895,52 @@ void reportProcedureDone(ReportProcedureDoneRequest request) throws IOException
     }
   }
 
+  /**
+   * Will ignore the open/close region procedures which already submitted or executed.
+   *
+   * When master had unfinished open/close region procedure and restarted, new active master may
+   * send duplicate open/close region request to regionserver. The open/close request is submitted
+   * to a thread pool and execute. So first need a cache for submitted open/close region procedures.
+   *
+   * After the open/close region request executed and report region transition succeed, cache it in
+   * executed region procedures cache. See {@link #finishRegionProcedure(long)}. And the cache will
+   * expire after 600 seconds. Because the duplicate open/close region request should not be delayed
```

Comment:
'Because the duplicated open/close region request is not likely to be delayed...'

And maybe we should also do something at master side? When reportRegionStateTransition succeeded, let's also remove the pending operations in RSProcedureDispatcher if possible? Or maybe add some checks in splitAndResolveOperation, if a RemoteProcedure has already been marked as succeeded, just give up and do not schedule it again. Otherwise we may still got duplicate requests if RSProcedureDispatcher keep retrying more than 10 minutes? Not likely but theoretically it is possible...